packages/@aws-cdk/aws-s3-notifications/lib/destination.ts

```diff
@@ -1,15 +1,18 @@
 import cdk = require('@aws-cdk/cdk');
-import { Bucket } from './bucket';
 
 /**
  * Implemented by constructs that can be used as bucket notification destinations.
  */
 export interface IBucketNotificationDestination {
     /**
-     * Registers this resource to receive notifications for the specified bucket.
-     * @param bucket The bucket. Use the `path` of the bucket as a unique ID.
+     * Registers this resource to receive notifications for the specified
+     * bucket. This method will only be called once for each destination/bucket
+     * pair and the result will be cached, so there is no need to implement
+     * idempotency in each destination.
+     * @param bucketArn The ARN of the bucket
+     * @param bucketId A unique ID of this bucket in the stack
      */
-    asBucketNotificationDestination(bucket: Bucket): BucketNotificationDestinationProps;
+    asBucketNotificationDestination(bucketArn: cdk.Arn, bucketId: string): BucketNotificationDestinationProps;
```

Comment:
We seem to have a inconsistent way of referring to resources throughout the CDK. Some use refs, some use Arns. Is it worth standardizing on a mechanism?

Also, don't we use "name" everywhere else to refer to the stack id? I like stack id better, but consistency.

Reply:
This is a special case since this library is a very low-level dependency and we can't use `BucketRef` here (because @aws-cdk/aws-s3 consumes this library). This is not too bad because this is an API would normally not interact with directly.

The idiomatic way is `BucketRef`

Reply:
This is not the name of the construct but rather a unique ID that represents the construct in the tree. I know it's a bit confusing, but it's a result of the low-levelness of this API. Otherwise, we would have just passed in the construct, and the user would be able to obtain this unique ID if they needed it to implement idempotency.

Reply:
An implementation comment (`//` instead of `/**`) might be helpful here to add that as a note.